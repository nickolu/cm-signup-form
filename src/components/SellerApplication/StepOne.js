import {useState} from 'react';
import {
    Box,
    FormControl,
    FormControlLabel,
    FormLabel,
    InputLabel,
    MenuItem,
    Radio,
    RadioGroup,
    Select,
    TextField,
} from '@mui/material';

export default function StepOne() {
    return (
        <Box
            component="form"
            onSubmit={(e) => {
                e.preventDefault();
            }}
        >
            <TextField id="first-name" label="First Name" />
            <TextField id="last-name" label="Last Name" />
            <FormControl>
                <InputLabel id="your-shop-category-label">
                    Your Shop Category
                </InputLabel>
                <Select
                    labelId="your-shop-category-label"
                    id="your-shop-category"
                    value={0}
                    label="Your Shop Category"
                    onChange={() => {}}
                >
                    <MenuItem value={0}>0</MenuItem>
                    <MenuItem value={1}>1</MenuItem>
                    <MenuItem value={2}>2</MenuItem>
                </Select>
            </FormControl>
            <TextField id="portfolio-link" label="Portfolio Link" />
            <FormControl component="fieldset">
                <FormLabel component="legend">
                    Do you already have an online store?
                </FormLabel>
                <RadioGroup
                    aria-label="online store"
                    defaultValue="yes"
                    name="have-online-store"
                >
                    <FormControlLabel
                        value="yes"
                        control={<Radio disableRipple />}
                        label="Yes"
                    />
                    <FormControlLabel
                        value="no"
                        control={<Radio disableRipple />}
                        label="No"
                    />
                </RadioGroup>
            </FormControl>
        </Box>
    );
}
