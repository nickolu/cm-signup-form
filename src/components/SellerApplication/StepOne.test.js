import {render, screen, within} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import StepOne from './StepOne';

describe('Step One of Signup Application', () => {
    let component;
    beforeEach(() => {
        component = render(<StepOne />);
    });

    it('should not have changed since last render', () => {
        const {container} = component;

        expect(component).toMatchSnapshot();
    });

    it('should render all of the form elements', () => {
        const {
            firstNameInput,
            lastNameInput,
            categorySelect,
            portfolioLinkInput,
            radioGroup,
            yesRadioButton,
            noRadioButton,
        } = queryformElements(screen);

        expect(firstNameInput).toBeInTheDocument();
        expect(lastNameInput).toBeInTheDocument();
        expect(categorySelect).toBeInTheDocument();
        expect(portfolioLinkInput).toBeInTheDocument();
        expect(radioGroup).toBeInTheDocument();
        expect(yesRadioButton).toBeInTheDocument();
        expect(noRadioButton).toBeInTheDocument();
    });

    test('first name input', () => {
        const {firstNameInput} = queryformElements(screen);

        userEvent.type(firstNameInput, 'Nick');

        const queriedInput = screen.queryByDisplayValue('Nick');

        expect(queriedInput).toBeInTheDocument();
    });

    test('last name input', () => {
        const {lastNameInput} = queryformElements(screen);

        userEvent.type(lastNameInput, 'Cunningham');

        const queriedInput = screen.queryByDisplayValue('Cunningham');

        expect(queriedInput).toBeInTheDocument();
    });

    test('category select', () => {
        const {categorySelect} = queryformElements(screen);

        userEvent.click(categorySelect);

        const firstElement = screen.getByLabelText('0');

        userEvent.click(firstElement);

        const queriedInput = screen.getByLabelText('0', {selected: true});

        expect(queriedInput).toBeInTheDocument();
    });

    test('portfolio link', () => {
        const {portfolioLinkInput} = queryformElements(screen);

        userEvent.type(portfolioLinkInput, 'mywebsite.com');

        const queriedInput = screen.queryByDisplayValue('mywebsite.com');

        expect(queriedInput).toBeInTheDocument();
    });

    test('radio group', () => {
        const {yesRadioButton, noRadioButton} = queryformElements(screen);

        userEvent.click(yesRadioButton);

        expect(yesRadioButton.checked).toBe(true);

        userEvent.click(noRadioButton);

        expect(noRadioButton.checked).toBe(true);
    });
});

function queryformElements(screen) {
    /*
        Note: using the accessible roles as selectors (instead of hard-coded ids) ensures that the app is accessible
        https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles
    */

    const firstNameInput = screen.queryByRole('textbox', {
        name: 'First Name',
    });
    const lastNameInput = screen.queryByRole('textbox', {
        name: 'Last Name',
    });
    const categorySelect = screen.getByLabelText('Your Shop Category');
    const portfolioLinkInput = screen.queryByRole('textbox', {
        name: 'Portfolio Link',
    });
    const radioGroup = screen.queryByRole('radiogroup');
    const yesRadioButton = screen.queryByRole('radio', {name: 'Yes'});
    const noRadioButton = screen.queryByRole('radio', {name: 'No'});

    return {
        firstNameInput,
        lastNameInput,
        categorySelect,
        portfolioLinkInput,
        radioGroup,
        yesRadioButton,
        noRadioButton,
    };
}
