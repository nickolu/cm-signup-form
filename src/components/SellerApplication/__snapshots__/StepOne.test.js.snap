// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step One of Signup Application should not have changed since last render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="MuiBox-root css-0"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="first-name"
            id="first-name-label"
          >
            First Name
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="first-name"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-173wfxe"
              >
                <span>
                  First Name
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="last-name"
            id="last-name-label"
          >
            Last Name
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="last-name"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-173wfxe"
              >
                <span>
                  Last Name
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            id="your-shop-category-label"
          >
            Your Shop Category
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="your-shop-category-label your-shop-category"
              class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              id="your-shop-category"
              role="button"
              tabindex="0"
            >
              0
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              tabindex="-1"
              value="0"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1z7n62"
              >
                <span>
                  Your Shop Category
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="portfolio-link"
            id="portfolio-link-label"
          >
            Portfolio Link
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="portfolio-link"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-173wfxe"
              >
                <span>
                  Portfolio Link
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <fieldset
          class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <legend
            class="MuiFormLabel-root MuiFormLabel-colorPrimary css-u4tvz2-MuiFormLabel-root"
          >
            Do you already have an online store?
          </legend>
          <div
            aria-label="online store"
            class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
            role="radiogroup"
          >
            <label
              class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
            >
              <span
                class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root Mui-checked css-vqmohf-MuiButtonBase-root-MuiRadio-root"
              >
                <input
                  checked=""
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  name="have-online-store"
                  type="radio"
                  value="yes"
                />
                <span
                  class="css-hyxlzm"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                    data-testid="RadioButtonUncheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-11zohuh-MuiSvgIcon-root"
                    data-testid="RadioButtonCheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </span>
              </span>
              <span
                class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
              >
                Yes
              </span>
            </label>
            <label
              class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
            >
              <span
                class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
              >
                <input
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  name="have-online-store"
                  type="radio"
                  value="no"
                />
                <span
                  class="css-hyxlzm"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                    data-testid="RadioButtonUncheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                    data-testid="RadioButtonCheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </span>
              </span>
              <span
                class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
              >
                No
              </span>
            </label>
          </div>
        </fieldset>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="MuiBox-root css-0"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="first-name"
          id="first-name-label"
        >
          First Name
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="first-name"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-173wfxe"
            >
              <span>
                First Name
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="last-name"
          id="last-name-label"
        >
          Last Name
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="last-name"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-173wfxe"
            >
              <span>
                Last Name
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          id="your-shop-category-label"
        >
          Your Shop Category
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="your-shop-category-label your-shop-category"
            class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="your-shop-category"
            role="button"
            tabindex="0"
          >
            0
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            tabindex="-1"
            value="0"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1z7n62"
            >
              <span>
                Your Shop Category
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="portfolio-link"
          id="portfolio-link-label"
        >
          Portfolio Link
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="portfolio-link"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-173wfxe"
            >
              <span>
                Portfolio Link
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <fieldset
        class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
      >
        <legend
          class="MuiFormLabel-root MuiFormLabel-colorPrimary css-u4tvz2-MuiFormLabel-root"
        >
          Do you already have an online store?
        </legend>
        <div
          aria-label="online store"
          class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
          role="radiogroup"
        >
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          >
            <span
              class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root Mui-checked css-vqmohf-MuiButtonBase-root-MuiRadio-root"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input css-1m9pwf3"
                name="have-online-store"
                type="radio"
                value="yes"
              />
              <span
                class="css-hyxlzm"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                  data-testid="RadioButtonUncheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-11zohuh-MuiSvgIcon-root"
                  data-testid="RadioButtonCheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </span>
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              Yes
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          >
            <span
              class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
            >
              <input
                class="PrivateSwitchBase-input css-1m9pwf3"
                name="have-online-store"
                type="radio"
                value="no"
              />
              <span
                class="css-hyxlzm"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                  data-testid="RadioButtonUncheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                  data-testid="RadioButtonCheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </span>
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              No
            </span>
          </label>
        </div>
      </fieldset>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
